rust/agama-lib/src/network/client.rs

```diff
@@ -98,32 +68,26 @@ impl NetworkClient {
         let response = self.connection(id.as_str()).await;
 
         if response.is_ok() {
-            let path = format!("{API_URL}/connections/{id}");
-            self.client
-                .put(path)
-                .json(&connection)
-                .send()
-                .await
-                .map_err(|e| ServiceError::NetworkClientError(e.to_string()))?;
+            let path = format!("/network/connections/{id}");
+            self.client.put_void(&path.as_str(), &connection).await?
         } else {
             self.client
-                .post(format!("{API_URL}/connections").as_str())
-                .json(&connection)
-                .send()
-                .await
-                .map_err(|e| ServiceError::NetworkClientError(e.to_string()))?;
+                .post_void(format!("/network/connections").as_str(), &connection)
+                .await?
         }
 
         Ok(())
     }
 
     /// Returns an array of network connections
     pub async fn apply(&self) -> Result<(), ServiceError> {
+        // trying to be tricky here. If something breaks then we need a put method on
+        // BaseHTTPClient which doesn't require a serialiable object for the body
+        let empty_body: [String; 0] = [];
```

Comment:
You may be interested in a shorter way to produce a dummy body:
https://github.com/agama-project/agama/commit/56335baf9e4ad8eeae78372c1a99ebe8696a6ec3